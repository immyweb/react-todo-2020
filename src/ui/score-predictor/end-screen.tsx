import * as React from 'react';
import glamorous from 'glamorous';

import Button from '../buttons/buttons';
import { mediaQueries } from '../../styles/breakpoints';
import { Copy } from './styles';
import { colours } from '../../styles/colours';
import { transparentize } from 'polished';
import { fonts } from '../../styles/fonts';

interface IProps {
  onCloseScreen(): void;
  onGoToUrl(): void;
}

export const EndScreen: React.SFC<IProps> = ({ onCloseScreen, onGoToUrl }) => {
  return (
    <EndScreenContainer>
      <EndScreenInner>
        <Image
          src="https://www.thesun.co.uk/wp-content/uploads/2018/03/ianwright.png"
          alt="The Sun Six"
        />
        <Copy>
          Get six scores right
          <br />
          this weekend &amp;
        </Copy>
        <Copy fontsize={34} fontbold={true} lineheight={1.2}>
          WIN Â£1 MILLION
        </Copy>
        <Button background={colours.$football} fullWidth={true} onClick={onGoToUrl}>
          Play now for free
        </Button>
        <Link onClick={onCloseScreen}>No thanks</Link>
      </EndScreenInner>
    </EndScreenContainer>
  );
};

const EndScreenContainer = glamorous.div({
  label: 'end-screen-container',
  width: '100vw',
  height: '100vh',
  background: transparentize(0.5, colours.$black),
  top: 0,
  left: 0,
  textAlign: 'center',
  position: 'fixed',
  zIndex: 17,
});

const EndScreenInner = glamorous.div({
  label: 'end-screen-inner',
  width: '90%',
  height: 'auto',
  background: colours.$white,
  padding: '10px 35px',
  textAlign: 'center',
  position: 'absolute',
  top: '50%',
  left: '50%',
  transform: 'translate(-50%, -50%)',
  zIndex: 16,
  [mediaQueries.tabletPortrait]: {
    width: 360,
  },
});

const Image = glamorous.img({
  label: 'end-screen-image',
  margin: '0 auto',
});

export const Link = glamorous.a({
  label: 'end-screen-no',
  color: colours.$football,
  fontFamily: fonts.$theSunMedium,
  marginTop: 10,
  marginBottom: 5,
  display: 'block',
  fontSize: 16,
  ':hover': {
    textDecoration: 'underline',
    cursor: 'pointer',
  },
});
