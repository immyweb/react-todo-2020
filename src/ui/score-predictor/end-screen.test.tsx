import * as React from 'react';
import { shallow } from 'enzyme';
import { EndScreen, Link } from './end-screen';
import Button from '../buttons/buttons';

describe('<EndScreen />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<EndScreen onCloseScreen={() => {}} onGoToUrl={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it(`onCloseScreen is called when "No thanks" button is clicked`, () => {
    const fn = jest.fn();
    const wrapper = shallow(<EndScreen onCloseScreen={fn} onGoToUrl={() => {}} />);
    const btn = wrapper.find(Link);
    btn.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
  });

  it(`goToUrl is called when "Play now for free" button is clicked`, () => {
    const fn = jest.fn();
    const wrapper = shallow(<EndScreen onCloseScreen={() => {}} onGoToUrl={fn} />);
    const btn = wrapper.find(Button);
    btn.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
  });
});
