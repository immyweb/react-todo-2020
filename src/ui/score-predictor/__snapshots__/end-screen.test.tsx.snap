// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EndScreen /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    onCloseScreen={[Function]}
    onGoToUrl={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <glamorous(div)>
        <glamorous(img)
          alt="The Sun Six"
          src="https://www.thesun.co.uk/wp-content/uploads/2018/03/ianwright.png"
        />
        <glamorous(p)>
          Get six scores right
          <br />
          this weekend &
        </glamorous(p)>
        <glamorous(p)
          fontbold={true}
          fontsize={34}
          lineheight={1.2}
        >
          WIN £1 MILLION
        </glamorous(p)>
        <glamorous(button)
          background="#459977"
          fullWidth={true}
          onClick={[Function]}
        >
          Play now for free
        </glamorous(button)>
        <glamorous(a)
          onClick={[Function]}
        >
          No thanks
        </glamorous(a)>
      </glamorous(div)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <glamorous(img)
            alt="The Sun Six"
            src="https://www.thesun.co.uk/wp-content/uploads/2018/03/ianwright.png"
          />,
          <glamorous(p)>
            Get six scores right
            <br />
            this weekend &
          </glamorous(p)>,
          <glamorous(p)
            fontbold={true}
            fontsize={34}
            lineheight={1.2}
          >
            WIN £1 MILLION
          </glamorous(p)>,
          <glamorous(button)
            background="#459977"
            fullWidth={true}
            onClick={[Function]}
          >
            Play now for free
          </glamorous(button)>,
          <glamorous(a)
            onClick={[Function]}
          >
            No thanks
          </glamorous(a)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": "The Sun Six",
            "src": "https://www.thesun.co.uk/wp-content/uploads/2018/03/ianwright.png",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Get six scores right",
              <br />,
              "this weekend &",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Get six scores right",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "this weekend &",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "WIN £1 MILLION",
            "fontbold": true,
            "fontsize": 34,
            "lineheight": 1.2,
          },
          "ref": null,
          "rendered": "WIN £1 MILLION",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "background": "#459977",
            "children": "Play now for free",
            "fullWidth": true,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Play now for free",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "No thanks",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "No thanks",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <glamorous(div)>
          <glamorous(img)
            alt="The Sun Six"
            src="https://www.thesun.co.uk/wp-content/uploads/2018/03/ianwright.png"
          />
          <glamorous(p)>
            Get six scores right
            <br />
            this weekend &
          </glamorous(p)>
          <glamorous(p)
            fontbold={true}
            fontsize={34}
            lineheight={1.2}
          >
            WIN £1 MILLION
          </glamorous(p)>
          <glamorous(button)
            background="#459977"
            fullWidth={true}
            onClick={[Function]}
          >
            Play now for free
          </glamorous(button)>
          <glamorous(a)
            onClick={[Function]}
          >
            No thanks
          </glamorous(a)>
        </glamorous(div)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(img)
              alt="The Sun Six"
              src="https://www.thesun.co.uk/wp-content/uploads/2018/03/ianwright.png"
            />,
            <glamorous(p)>
              Get six scores right
              <br />
              this weekend &
            </glamorous(p)>,
            <glamorous(p)
              fontbold={true}
              fontsize={34}
              lineheight={1.2}
            >
              WIN £1 MILLION
            </glamorous(p)>,
            <glamorous(button)
              background="#459977"
              fullWidth={true}
              onClick={[Function]}
            >
              Play now for free
            </glamorous(button)>,
            <glamorous(a)
              onClick={[Function]}
            >
              No thanks
            </glamorous(a)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "The Sun Six",
              "src": "https://www.thesun.co.uk/wp-content/uploads/2018/03/ianwright.png",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Get six scores right",
                <br />,
                "this weekend &",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Get six scores right",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              "this weekend &",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "WIN £1 MILLION",
              "fontbold": true,
              "fontsize": 34,
              "lineheight": 1.2,
            },
            "ref": null,
            "rendered": "WIN £1 MILLION",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "background": "#459977",
              "children": "Play now for free",
              "fullWidth": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Play now for free",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "No thanks",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "No thanks",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
